File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_think/20260910_094726/typescript/008_prime_numbers/solution.ts
/**
 * The program finds all prime numbers up to and including N.
 * It uses the Sieve of Eratosthenes algorithm for efficiency.
 */

function main() {
    // Get the argument N from the command line
    const arg = process.argv[2];

    if (arg === undefined) {
        return;
    }

    const N = parseInt(arg, 10);

    // If N is not a number or is less than 2, there are no primes to print
    if (isNaN(N) || N < 2) {
        return;
    }

    // Sieve of Eratosthenes algorithm
    // Create a boolean array "isPrime[0..N]" and initialize
    // all entries it as true. A value in isPrime[i] will
    // finally be false if i is Not a prime, else true.
    const isPrime = new Uint8Array(N + 1).fill(1);
    isPrime[0] = 0;
    isPrime[1] = 0;

    for (let p = 2; p * p <= N; p++) {
        // If isPrime[p] is not changed, then it is a prime
        if (isPrime[p] === 1) {
            // Update all multiples of p starting from p*p
            // because smaller multiples have already been marked
            for (let i = p * p; i <= N; i += p) {
                isPrime[i] = 0;
            }
        }
    }

    // Collect and print all prime numbers
    let output = "";
    for (let p = 2; p <= N; p++) {
        if (isPrime[p] === 1) {
            output += p + "\n";
        }
        
        // Periodically flush the output string to avoid memory issues 
        // with extremely large N and very long strings
        if (output.length > 8192) {
            process.stdout.write(output);
            output = "";
        }
    }
    process.stdout.write(output);
}

main();

File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_think/20260910_094726/python/008_prime_numbers/solution.py
import sys

def is_prime(num):
    """Checks if a number is prime."""
    if num < 2:
        return False
    # Check divisors from 2 up to the square root of the number
    for i in range(2, int(num**0.5) + 1):
        if num % i == 0:
            return False
    return True

def main():
    # Ensure there is at least one argument passed via command line
    if len(sys.argv) < 2:
        return

    try:
        # Parse the first command line argument as an integer
        n = int(sys.argv[1])
    except ValueError:
        # If the argument is not a valid integer, exit silently
        return

    # Iterate through all numbers from 2 up to and including N
    for i in range(2, n + 1):
        if is_prime(i):
            print(i)

if __name__ == "__main__":
    main()